src/node_file.cc

```diff
@@ -1270,8 +1270,19 @@ int MKDirpSync(uv_loop_t* loop,
         }
         default:
           uv_fs_req_cleanup(req);
+          if (err == UV_EEXIST && continuation_data.paths.size() > 0) {
```

Comment:
Nit: Would it be better to move `UV_EEXIST` to a `case`?

Reply:
We'd still need to duplicate the logic for when `continuation_data.paths.size() == 0` (which is the same logic as the rest of the default case).

Reply:
Given that we now have logic that checks whether or not a path is a file before calling `MKDirpAsync`, do we need to check `continuation_data.paths.size() > 0`; seems like a reasonable safety to have in place, but wonder if you could think of a situation where we'd hit this edge-case.

Reply:
We haven't done the check the very first time `MKDirpAsync` is called from `MKDir`. So if `uv_fs_mkdir()` fails on the very first iteration (`continuation_data.paths.size() == 0`) we want to return `UV_EEXIST`.